tests/ops/qutrit/test_qutrit_non_parametric_ops.py

```diff
@@ -70,11 +72,19 @@ def test_tclock_eigenval(self):
         res = op.eigvals()
         assert np.allclose(res, exp)
 
+    def test_tadd_eigenval(self):
```

Comment:
There should be a similar test for `TSWAP` ?

Reply:
`TSWAP` doesn't have an `eigvals` method. If I'm being completely honest, I'm not entirely sure that `TAdd` or any of the other non-parametric operations even need an `eigvals` method. I'd love to get some input on this.

Reply:
Just looking at the non-parametric qubit operations, I can't actually figure out why some of the operations have eigvals and some don't; I thought it might be diagonal vs. non-diagonal, or things that are also observables, but that doesn't seem to be the case. @josh146 do you know the reason for this?

Reply:
I agree that we don't directly use it for any measurements, but I think asking for the eigenvalues of an operator is a very natural thing. Almost as natural as asking for the matrix representation of the operator. I think we should definitely support it for non-parametric operators where the result is well known (no dependance on external parameters).  

Also, with operator arithmetic, one will have the ability to combine these operators to create observables which can then be measured. We definitely make use of the Eigen values when performing these measurements.

Reply:
I think we only ever added eigvals consistently to observables, since they are used during expectation value computation. I don't think we internally use eigvals anywhere if they are provided by a non-observable operation